cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -524,21 +529,39 @@ public RunId runInternal(ProgramId programId, Map<String, String> userArgs, Map<
     userId = userId == null ? "" : userId;
 
     checkCapability(programDescriptor);
-    synchronized (this) {
-      if (maxConcurrentRuns > 0 && maxConcurrentRuns <= store.countActiveRuns(maxConcurrentRuns)) {
-        ConflictException e = new ConflictException(
-          String.format("Program %s cannot start because the maximum of %d concurrent runs is exceeded",
-                        programId, maxConcurrentRuns));
 
+    RunRecordCounter.Count count = runRecordCounter.addRequestAndGetCount(programId.run(runId));
+
+      if (maxConcurrentRunning > 0 && maxConcurrentRunning < count.getRunningCount()) {
```

Comment:
Misaligned block.

Reply:
Should be `>= 0`, since the doc mentioned setting it to negative to disable the check.